gnosis/protocol/gnosis_protocol_api.py

```diff
@@ -96,10 +99,20 @@ def get_fee(self, order: Order, from_address: ChecksumAddress) -> int:
         }
         r = self.http_session.post(url, json=data_json)
         if r.ok:
-            return int(r.json()["quote"]["feeAmount"])
+            return r.json()
         else:
             return ErrorResponse(r.json())
 
+    def get_fee(
+        self, order: Order, from_address: ChecksumAddress
+    ) -> Union[int, ErrorResponse]:
+        quote = self.get_quote(order, from_address)
+
+        if "quote" in quote:
+            return int(quote["quote"]["feeAmount"])
+        else:
+            return quote
```

Comment:
here we are returning an error?

Reply:
Yes, we should use exceptions, I don't know why I did it this way (like Golang), but maybe it's too much to refactor everything now